purchase_work_acceptance/models/stock_picking.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2019 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class Picking(models.Model):
+    _inherit = "stock.picking"
+
+    require_wa = fields.Boolean(
+        compute='_compute_require_wa',
+    )
+    wa_id = fields.Many2one(
+        comodel_name='work.acceptance',
+        string='WA Reference',
+        copy=False,
+        domain=lambda self: [
+            ('state', '=', 'accept'),
+            ('purchase_id', '=', self._context.get('active_id'))],
+    )
+
+    @api.multi
+    def _compute_require_wa(self):
+        self.require_wa = self.env.user.has_group(
+            'purchase_work_acceptance.group_work_acceptance_enforce')
+
+    @api.multi
+    def button_validate(self):
+        if self.wa_id:
+            wa_line = {}
+            for line in self.wa_id.wa_line_ids:
+                qty = line.product_uom._compute_quantity(
+                    line.product_qty, line.product_id.uom_id)
+                if qty > 0.0 and line.product_id.type != 'service':
```

Comment:
you should make `type in ['product', 'consu']` as this is the standard logic of picking.

Reply:
Edited : https://github.com/OCA/purchase-workflow/pull/774/commits/74b8c5f27338d714143dfee5372105bf91e269cc